docker-compose-rule-core/src/main/java/com/palantir/docker/compose/configuration/ShutdownStrategy.java

```diff
@@ -60,6 +59,8 @@ static ShutdownStrategy callbackAndThen(Runnable callback, ShutdownStrategy shut
         return new CallbackThenDelegateShutdownStrategy(shutdownStrategy, callback);
     }
 
-    void shutdown(DockerCompose dockerCompose, Docker docker) throws IOException, InterruptedException;
```

Comment:
Removing this is a break as `ShutdownStrategy` is public api. Can we keep `shutdown(DockerCompose, Docker)` as a default method (and deprecate it), then call out to `stop` and `down` which are default empty methods (but implemented by everything in the main library?)? That way if anyone has implemented their own shutdown strategy they will not be broken?

Reply:
Actually, that might not work as we can't just call shutdown, we want to call `stop` then `down`. Maybe keeping an empty deprecated default `shutdown(...)` method and calling it from `stop(...)` would be the actual behaviour we want?

Reply:
The current implementation of `shutdown` is effectively just `docker-compose down` (with some implementations calling `docker-compose stop` first to give processes a chance to end cleanly). I think it's better to preserve the existing behavior with the downside that you don't get logs if you call `shutdown` manually (but you really shouldn't be doing that anyways).

Reply:
The thing is I don't think people will be calling `shutdown` manually - it's DCR itself executing the `ShutdownStrategy`, so after this change if someone had written a custom `ShutdownStrategy` and gives it to DCR nothing will happen. Hence why we need to make `stop()` or `down()` call `shutdown()`, so the user's old shutdown code still works.

Reply:
If a user wrote a custom `ShutdownStrategy` it would no longer compile because they don't implement the `stop()` and `down()` methods.

Reply:
That's what I'm saying we should try to avoid - people should not experience a break from this change, if possible. They can be made default methods to not cause breaks.

Reply:
Changing `down()` to call `shutdown()` requires also passing in `Docker` which I'd like to avoid (since you shouldn't need to use docker directly to shut down). Instead let's just continue to call `shutdown()` directly and make it a no-op default implementation.